executor/builder.go

```diff
@@ -3527,10 +3527,14 @@ func getPartitionKeyColOffsets(keyColIDs []int64, pt table.PartitionedTable) []i
 		keyColOffsets[i] = offset
 	}
 
-	pe, err := pt.(interface {
-		PartitionExpr() (*tables.PartitionExpr, error)
-	}).PartitionExpr()
-	if err != nil {
+	t, ok := pt.(interface {
```

Comment:
(For now)Is it possible that `ok` is false?

Reply:
No, it should not be possible currently, but I think it is better to check anyway :)